robot/impl/local_robot.go

```diff
@@ -603,6 +609,51 @@ func (r *localRobot) getWeakDependencyMatchers(api resource.API, model resource.
 	return reg.WeakDependencies
 }
 
+func (r *localRobot) getOptionalDependencies(conf resource.Config) resource.Dependencies {
+	optDeps := make(resource.Dependencies)
+
+	for _, optionalDepNameString := range conf.ImplicitOptionalDependsOn {
+		matchingResourceNames := r.manager.resources.FindNodesByShortName(optionalDepNameString)
+		switch len(matchingResourceNames) {
+		case 0:
+			r.logger.Infow(
```

Comment:
This log and the log on L643 will appear like this:

```
2025-05-01T18:05:35.587Z	INFO	rdk	impl/local_robot.go:630	Optional dependency for resource does not exist; not passing to constructor or reconfigure yet	{"dependency":"m2","resource":"rdk:component:generic/f"}
```

And are helpful to distinguish _why_ an optional dependency is _not_ being passed to a constructor or reconfigure method. If the optional dependency is valid (not a self-dependency and not ambiguous due to multiple matches), then it is either the case that there is no associated resource for the dependency, or the associated resource is not available for some reason. The logs I've added should help distinguish between those cases.